src/js/components/Calendar/Calendar.js

```diff
@@ -40,7 +40,14 @@ const headingPadMap = {
 const buildStartEnd = (reference, firstDayOfWeek) => {
   let start = new Date(reference);
   start.setDate(1); // first of month
-  start = subtractDays(start, start.getDay() - firstDayOfWeek); // beginning of week
+
+  // in case Sunday is the first day of the months and the user asked for Monday to be the first day of the week, we need to include Sunday and six days prior.
```

Comment:
```suggestion
  // in case Sunday is the first day of the month, and the user asked for Monday to be the first day of the week, we need to include Sunday and six days prior.
```